### PR TITLE
persons: handle departments m2m

### PR DESCRIPTION
**Pull request by HugoDelval**
**Problem**

The back weren't ready for this PR : https://github.com/cgwire/kitsu/pull/641

The back wasn't accepting the `departments` key in update and create. It dod not returns the list of departments in the serializers

**Solution**

- modify serializers to include the departments
- modify the update/create views to adds the departments in the `Person` object

### CODE REVIEW

File: zou/app/blueprints/crud/person.py [hunk 1]
```diff
@@ -88,6 +92,11 @@ def post_delete(self, instance_dict):
     def update_data(self, data, instance_id):
         if "password" in data:
             del data["password"]
+        if "departments" in data:
+            departments = [
+                Department.get(department_id) for department_id in data["departments"]
```
Comment (frankrousseau) [inline zou/app/blueprints/crud/person.py:97]: It doesn't handle the case where the department_id matches nothing in the database.

File: zou/app/blueprints/crud/person.py [hunk 2]
```diff
@@ -22,11 +23,11 @@ def all_entries(self, query=None, relations=False):
 
         if permissions.has_manager_permissions():
             if request.args.get("with_pass_hash") == "true":
-                return [person.serialize() for person in query.all()]
+                return [person.serialize(relations=True) for person in query.all()]
             else:
-                return [person.serialize_safe() for person in query.all()]
+                return [person.serialize_safe(relations=True) for person in query.all()]
```
Comment (frankrousseau) [inline zou/app/blueprints/crud/person.py:28]: Can you make lines with length < 80 chars? I'm not sure that this one is not too wide.

**Review by frankrousseau — COMMENTED** [review]
Thank you for this PR! I added a few remarks to improve the code.

File: zou/app/blueprints/crud/person.py (hunk 1 above)
Comment (HugoDelval) [reply zou/app/blueprints/crud/person.py]: It now raises a 404 exception if one department_id is not present in the DB

**Comment by HugoDelval** [comment]
I've also added some tests for adding/updating a person with department

**Comment by frankrousseau** [comment]
Awesome, thx!

**Pull request merged**